pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -301,6 +304,10 @@ func (self *AWSCloud) AddSSHKeyToAllInstances(user string, keyData []byte) error
 }
 
 func (a *AWSCloud) CurrentNodeName(hostname string) (string, error) {
+	if a.cfg.Global.UseKubernetesNodeTag {
```

Comment:
Can we just use the KubernetesNodeTag here, rather than assuming that hostname == KubernetesNodeTag?  We're about to get it anyway when we call getSelfAWSInstance